src/compressionworker.ts

```diff
@@ -0,0 +1,105 @@
+import compress from "./compress";
+import { config } from "./config";
+
+export function createCompressionWorker(
+  envelope: IEnvelope,
+  onMessage?: (e: MessageEvent) => void,
+  onError?: (e: ErrorEvent) => void
+): Worker {
+  let workerUrl = createWorkerUrl(envelope);
+  let worker = new Worker(workerUrl);
+  worker.onmessage = onMessage || null;
+  worker.onerror = onError || null;
+  return worker;
+}
+
+function workerContext() {
+  let workerGlobalScope = self as any;
+  let compress = workerGlobalScope.compress;
+  let config = workerGlobalScope.config;
+  let envelope: IEnvelope = workerGlobalScope.envelope;
+  let nextPayloadEvents: IEvent[] = [];
+  let nextPayloadBytes = 0;
+  let sequence = 0;
+
+  // Edge case: Flag to skip uploading payloads consisting of a single XhrError instrumentation event
+  // This helps us avoid the infinite loop in the case when all requests fail (e.g. dropped internet connection)
```

Comment:
>infinite [](start = 29, length = 8)

Could we instead change the XHR callback to hit a maximum number of retires? This is separate from your changes.

Reply:
This infinite loop doesn't come from re-uploading dropped payloads. It comes from generating new event for a failed send and sending that as a new payload, so it's a different issue.

---
In reply to: [137239428](https://github.com/Microsoft/clarity-js/pull/28#discussion_r137239428) [](ancestors = 137239428)

Reply:
Understood it's a different issue. Just wondering if we should fix the "never ending retry" problem (not the infinite loop) if easy as part of this PR. 

---
In reply to: [137353755](https://github.com/Microsoft/clarity-js/pull/28#discussion_r137353755) [](ancestors = 137353755,137239428)

Reply:
I think that making a separate PR, which would split off the sending/status tracking/re-seding logic in a separate entity, would be a better fit for that change. (I was going to make another PR for that, regardless)

---
In reply to: [137421412](https://github.com/Microsoft/clarity-js/pull/28#discussion_r137421412) [](ancestors = 137421412,137353755,137239428)